qiita_pet/handlers/qiita_redbiom.py

```diff
@@ -0,0 +1,117 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from future.utils import viewitems
+from requests import ConnectionError
+from collections import defaultdict
+import redbiom.summarize
+import redbiom.search
+import redbiom._requests
+import redbiom.util
+import redbiom.fetch
+from tornado.gen import coroutine, Task
+
+from qiita_core.util import execute_as_transaction
+from qiita_db.util import generate_study_list_without_artifacts
+
+from .base_handlers import BaseHandler
+
+
+class RedbiomPublicSearch(BaseHandler):
+    @execute_as_transaction
+    def get(self, search):
+        self.render('redbiom.html')
+
+    @execute_as_transaction
+    def _redbiom_search(self, query, search_on, callback):
+        message = ''
+        results = []
+
+        try:
+            df = redbiom.summarize.contexts()
+        except ConnectionError:
+            message = 'Redbiom is down - contact admin, thanks!'
+
+        if not message:
+            study_artifacts = defaultdict(list)
+            contexts = df.ContextName.values
+            if search_on == 'metadata':
+                query = query.lower()
+                try:
+                    samples = redbiom.search.metadata_full(query, False)
+                except TypeError:
+                    message = (
+                        'Not a valid search: "%s", are you sure this is a '
+                        'valid metadata %s?' % (
+                            query, 'value' if search_on == 'metadata' else
+                            'category'))
+                except ValueError:
+                    message = (
+                        'Not a valid search: "%s", your query is too small '
+                        '(too few letters), try a longer query' % query)
+
+                if message == '':
```

Comment:
can you describe in english what the goal of this block is? it looks like the intention is to establish a dict mapping a study, to the artifacts contained in that study that overlap with requested samples? if yes, wouldn't it be easier to just query the qiita API for that?

Reply:
I guess I'm also a bit confused as to why the artifact IDs are necessary? I'm concerned with the subsequent code as it depends on the structure of redbiom IDs which we do not make guarantees about. If this is necessary (which I think can be handled by cross referencing with the Qiita API), can you please open an issue so that we can construct an API endpoint to retrieve the tags from a set of redbiom IDs?